core/storage/classifier/gae_models.py

```diff
@@ -44,70 +43,42 @@ class ClassifierModel(base_models.BaseModel):
     algorithm_id = ndb.StringProperty(required=True, choices=ALGORITHM_CHOICES)
     # The actual model used for classification. Immutable, unless a schema
     # upgrade takes place.
-    cached_classifier_data = ndb.JsonProperty(required=True)
+    classifier_data = ndb.JsonProperty(required=True)
     # The schema version for the data that is being classified.
     data_schema_version = ndb.IntegerProperty(required=True)
 
     @classmethod
-    def _generate_id(cls, exp_id):
-        """Generates a unique id for the classifier model of the form
-        {{exp_id}}.{{random_hash_of_16_chars}}
-
-        Args:
-            exp_id: str. ID of the exploration.
-
-        Returns:
-            ID of the new classifier model.
-
-        Raises:
-            Exception: The id generator for ClassifierModel is producing too
-            many collisions.
-        """
-
-        for _ in range(base_models.MAX_RETRIES):
-            new_id = '%s.%s' % (
-                exp_id,
-                utils.convert_to_hash(
-                    str(utils.get_random_int(base_models.RAND_RANGE)),
-                    base_models.ID_LENGTH))
-            if not cls.get_by_id(new_id):
-                return new_id
-
-        raise Exception(
-            'The id generator for ClassifierModel is producing too many '
-            'collisions.')
-
-    @classmethod
     def create(
-            cls, exp_id, exp_version_when_created, state_name, algorithm_id,
-            cached_classifier_data, data_schema_version):
-        """Creates a new ClassifierModel entry.
+            cls, job_request_id, exp_id, exp_version_when_created, state_name,
```

Comment:
It is not necessary to explicitly mention that we are using `job_request_id`. It can be named `classifier_id`. It does not need to know that it is same as `job_request_id`. We will pass `job_request_id` in this argument when saving a new model. So no need to use `job_request_id` explicitly.